package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -84,93 +89,143 @@ _proto_mbim_setup() {
 	echo "mbim[$$]" "Checking pin"
 	umbim $DBG -n -t $tid -d $device pinstate || {
 		echo "mbim[$$]" "PIN required"
-		proto_notify_error "$interface" PIN_FAILED
-		proto_block_restart "$interface"
-		return 1
+		if [ ! "$ignore_pinfail" ] || [ "$ignore_pinfail" = 0 ]; then
+			proto_notify_error "$interface" PIN_FAILED
+			proto_block_restart "$interface"
+			return 1
+		else
+			echo "mbim[$$]" "Ignoring PIN failure"
+		fi
 	}
 	tid=$((tid + 1))
 
 	echo "mbim[$$]" "Checking subscriber"
- 	umbim $DBG -n -t $tid -d $device subscriber || {
+	umbim $DBG -n -t $tid -d $device subscriber || {
```

Comment:
Whitespace cleanup is nice, but reviewing is simpler if that's kept in a separate patch